simpeg/optimization.py

```diff
@@ -1038,7 +1171,7 @@ def findSearchDirection(self):
         return Solver(self.H) * (-self.g)
 
 
-class InexactGaussNewton(BFGS, Minimize, Remember):
+class InexactGaussNewton(InexactCG, BFGS):
```

Comment:
In the future, I'd like to remove the BFGS Inheritance here...
Currently it is only used as a default preconditioner to the inner CG iterations, which is not necessarily a bad thing if the objective function doesn't change, but it's questionable if it is appropriate to continue using it when a directive changes the objective function, as the BFGS memory is not related to the current objective function. I'm not sure of the best approach just yet, but I don't think inheriting from BFGS just to get access to it's behavior for default options is a good approach.